src/plugins/opensearch_dashboards_react/public/table_list_view/edit_action_dropdown.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiContextMenu, EuiIcon, EuiPopover } from '@elastic/eui';
+import React, { useState } from 'react';
+
+//TODO: include more types once VisBuilder supports more visualization types
+const types = ['Area', 'Vertical Bar', 'Line', 'Metric', 'Table']
+
+interface Item {
+    typeTitle: string;
+    id?: string;
+    version?: number;
+  }
+  
+  interface EditActionDropdownProps {
+    item: Item;
+    editItem?(item: object): void;
+    visbuilderEditItem?(item: object): void;
+  }
+  
+  const EditActionDropdown: React.FC<EditActionDropdownProps> = ({
+    item,
+    editItem,
+    visbuilderEditItem,
+  }) => {
+    const [isPopoverOpen, setPopoverOpen] = useState(false);
+  
+    const onButtonClick = () => {
+      setPopoverOpen(!isPopoverOpen);
+    };
+  
+    const closePopover = () => {
+      setPopoverOpen(false);
+    };
+  
+    // A saved object will only have the 'Import to VisBuilder' option 
+    // if it is a VisBuilder-compatible type and its version is 1.
+    const typeName = item.typeTitle;
+    const itemVersion = item.version;
+    const isSpecificType = types.includes(typeName) && itemVersion && itemVersion <= 1;
+  
+    const panels = [
+      {
+        id: 0,
+        items: [
+          {
+            name: 'Edit',
+            icon: 'pencil',
+            onClick: () => {
+              closePopover();
+              editItem && editItem(item);
+            },
+          },
+          ...(isSpecificType
+            ? [
+                {
+                  name: 'Import to VisBuilder',
+                  icon: 'importAction',
+                  onClick: () => {
+                    closePopover();
+                    visbuilderEditItem && visbuilderEditItem(item);
+                  },
+                },
+              ]
+            : []),
+        ],
+      },
+    ];
+  
+    return (
+      <EuiPopover
+        button={<EuiIcon type="pencil" onClick={onButtonClick} />}
+        isOpen={isPopoverOpen}
+        closePopover={closePopover}
+        panelPaddingSize="none"
+        initialFocus={false}
+      >
+        <EuiContextMenu initialPanelId={0} panels={panels} />
+      </EuiPopover>
+    );
+  };
+  
+  export default EditActionDropdown;
```

Comment:
try to avoid default exports if no specific reason? i think most components don't use export default, unless they need to be lazy imported

Reply:
thanks, fixed